lib/sql-logging/statistics.rb

```diff
@@ -1,105 +1,119 @@
 module SqlLogging
-  class Helper
-    include Singleton
-    include ActionView::Helpers::TextHelper
-    include ActionView::Helpers::NumberHelper
-  end
-  
   class Statistics
-    @@show_sql_backtrace = true
-    @@show_top_sql_queries = :total_time
-    @@top_sql_queries = 10
-    @@backtrace_cleaner = nil
-  
-    cattr_accessor :show_sql_backtrace, :top_sql_queries
-  
-    def self.show_top_sql_queries
-      @@show_top_sql_queries
-    end
-  
-    def self.show_top_sql_queries=(value)
-      unless [ false, :rows, :queries, :bytes, :total_time, :median_time ].include?(value)
-        raise ArgumentError, "show_top_sql_queries must be one of false, :rows, :queries, :bytes, :total_time or :median_time"
+    class Data
+      COUNTERS = %i(queries bytes rows).freeze
+      attr_accessor(*COUNTERS)
+      attr_reader(:top_queries)
+
+      def initialize
+        reset!
       end
-    
-      @@show_top_sql_queries = value
-    end
 
-    @@queries = @@bytes = @@rows = 0
-    @@top_queries = {}
+      def reset!
+        COUNTERS.each { |v| instance_variable_set("@#{v}", 0) }
+        @top_queries = {}
+      end
 
-    def self.reset_statistics!
-      @@queries = @@bytes = @@rows = 0
-      @@top_queries = {}
+      def add_query(row_counts, bytes)
+        @queries += 1
+        @bytes += bytes
+        @rows += row_counts
+      end
     end
 
-    def self.backtrace_cleaner
-      unless @@backtrace_cleaner
-        @@backtrace_cleaner = Rails.backtrace_cleaner.dup
-        @@backtrace_cleaner.add_silencer { |line| line =~ %r{sql-logging/lib} }
+    class << self
+      extend Forwardable
+
+      def configuration
+        SqlLogging.configuration
+      end
+
+      def_delegators :configuration, :show_sql_backtrace, :top_sql_queries,
```

Comment:
That PR is already large. We could split it differently, but I'm not against keeping it in this format and iterate. We can keep this and discuss it afterward.